source/elements/oneDAL/include/oneapi/dal/table/column_accessor.hpp

```diff
@@ -0,0 +1,61 @@
+#include "onedal/table.hpp"
+
+namespace oneapi::dal {
+
+/// @tparam Data The type of data values in blocks returned by the accessor.
+///              Shall be const-qualified for read-only access.
+template <typename Data>
+class column_accessor {
+public:
+    using data_t = std::remove_const_t<Data>;
+
+public:
+    /// Creates a new read-only accessor object from the table.
+    /// The check that accessor supports the table kind of $obj$ shall be performed.
+    /// The reference on the $obj$ table shall be stored within the accessor to
+    /// perform data reading from it.
+    column_accessor(const table& obj);
+
+    /// Provides access to the column values of the table.
+    /// The method shall return the array that directly points to the memory within the table
+    /// if it is possible. In that case, the array shall refer to the memory as to immutable data.
+    /// Otherwise, the new memory block shall be allocated, data from the table rows shall be converted
+    /// and copied into this block. The array shall refer to the block as to mutable data.
```

Comment:
```suggestion
    /// The method shall return an array that directly points to the memory within the table
    /// if possible. In that case, the array shall refer to the memory as to immutable data.
    /// Otherwise, the new memory block shall be allocated, the data from the table rows shall be converted
    /// and copied into this block. The array shall refer to the block as to mutable data.
```